### PR TITLE
Two kinds of proposals

### PR DESCRIPTION
**Pull request by deep-quality-dev**
Check length of choices with transfer information

### CODE REVIEW

File: src/client/DAOClient.ts [hunk 1]
```diff
@@ -353,6 +354,17 @@ class DAOClient implements zDAO {
       throw new Error(errorMessageForError('invalid-proposal-duration'));
     }
 
+    if (payload.transfer) {
+      if (payload.choices && payload.choices.length !== 2) {
```
Comment (JamesEarle) [inline src/client/DAOClient.ts:358]: Consider simplifying this logic a bit by first checking `if (!payload.choices)`''

```typescript
if (payload.transfer) {
  if (!payload.choices) {
      payload.choices = DEFAULT_CHOICES;
  } else if (payload.choices.length !== 2) // we know payload.choices is valid here, no need for duplicative check
      throw new Error(...);
  }
}

**Review by JamesEarle — APPROVED** [review]

File: src/client/DAOClient.ts (hunk 1 above)
Comment (deep-quality-dev) [reply src/client/DAOClient.ts:358]: Thanks for your review, that makes sense

Comment (deep-quality-dev) [reply src/client/DAOClient.ts:358]: I am thinking if client didn't pass choices, then use `DEFAULT_CHOICES` by default.

File: src/types/params.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export interface CreateZDAOParams {
 export interface CreateProposalParams {
   title: string;
   body: string;
-  choices: string[];
+  choices?: string[];
```
Comment (Remscar) [inline src/types/params.ts:27]: I have a slight concern that somebody could manipulate the system by inputting the choices ["Deny", "Approve"],

Perhaps we should not allow choices to be passed for funding proposals, and default to "Approve", "Deny"

**Review by Remscar — APPROVED** [review]
Looks good

File: src/types/params.ts (hunk 2 above)
Comment (deep-quality-dev) [reply src/types/params.ts:27]: I think so, then I will left the message in Discord and get feedback from @Eth Fish and @Brett. 
If they also agree to use default choices(Approve/Deny) for funding proposal, then I will merge into main

**Pull request merged**